atom-jobs/cache-code.groovy

```diff
@@ -0,0 +1,229 @@
+/*
+* input
+* @REPO(string: github repo name, Required)
+* @PULL_ID(int: pull request id, Optional)
+* @COMMIT_ID(string: commit id request id, Required)
+* @BRANCH(string: branch name, Optional)
+*
+*
+* output
+* @CACHE_CODE_URL(string: cache code tar file download url)
+* @CACHE_CODE_SHA256_URL(string: cache code tar file sha256 sum file url)
+*
+* doc
+* https://docs.google.com/document/d/1hcs4F6BUKiiLHetLMXM2323_J5_yLdG5OyXddNt2-Vk/edit?usp=sharing
+*/
+
+properties([
+    parameters([
+        string(
+            defaultValue: '',
+            name: 'REPO',
+            trim: true
+        ),
+        string(
+            defaultValue: '',
+            name: 'PULL_ID',
+            trim: true
+        ),
+        string(
+            defaultValue: '',
+            name: 'COMMIT_ID',
+            trim: true,
+        ),
+        string(
+            defaultValue: '',
+            name: 'BRANCH',
+            trim: true
+        ),
+        booleanParam(
+            defaultValue: false,
+            name: 'FORCE_CACHE'
+        ),
+    ])
+])
+
+
+repoUrlMap = [
```

Comment:
should we hard code for this?

Reply:
ci service only support some specific repos which has been approved by ee team